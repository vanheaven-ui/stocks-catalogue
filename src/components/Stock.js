import { PropTypes } from 'prop-types';
import { Badge, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const Stock = ({ stock }) => (
  <>
    <div className="company-preview">
      <h2>
        { stock.name.length > 18 ? `${stock.name.substring(0, 15)}...` : stock.name }
      </h2>
      <h2><Badge variant="primary">{stock.symbol}</Badge></h2>
    </div>
    <hr />
    <div className="stock-stats">
      <h2>
        Price:
        {' '}
        { stock.price }
      </h2>
    </div>
    <div className="stock-footer">
      <Button
        variant="outline-primary"
        style={{ backgroundColor: '#007bff', color: '#fff' }}
      >
        <Link to={`/stocks/${stock.symbol}`}>Details</Link>
      </Button>
      <div className="exchange">
        <small style={{ color: '#d1450d' }}>Trading on:</small>
        <h6>{stock.exchange}</h6>
      </div>
    </div>
  </>
);

Stock.propTypes = {
  stock: PropTypes.instanceOf(Object).isRequired,
};

export default Stock;
