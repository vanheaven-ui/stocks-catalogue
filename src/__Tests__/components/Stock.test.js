import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Stock from '../../components/Stock';
import rootNode from '../helpers';

describe('Stock component', () => {
  const stock = {
    symbol: 'SPY', name: 'SPDR S&P 500', price: 395.98, exchange: 'New York Stock Exchange Arca',
  };

  test('renders without crashing', () => {
    ReactDOM.render(<BrowserRouter><Stock stock={stock} /></BrowserRouter>, rootNode());
  });

  test('doesnot change unexpectedly', () => {
    const tree = renderer.create(<BrowserRouter><Stock stock={stock} /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
