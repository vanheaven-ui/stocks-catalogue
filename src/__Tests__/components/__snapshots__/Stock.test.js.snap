// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock component doesnot change unexpectedly 1`] = `
Array [
  <div
    className="company-preview"
  >
    <h2>
      SPDR S&P 500
    </h2>
    <h2>
      <span
        className="badge badge-primary"
      >
        SPY
      </span>
    </h2>
  </div>,
  <hr />,
  <div
    className="stock-stats"
  >
    <h2>
      Price:
       
      395.98
    </h2>
  </div>,
  <div
    className="stockFooter"
  >
    <button
      className="btn btn-outline-primary"
      disabled={false}
      style={
        Object {
          "backgroundColor": "#007bff",
        }
      }
      type="button"
    >
      <a
        href="/stocks/SPY"
        onClick={[Function]}
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        Details
      </a>
    </button>
    <div
      className="exchange"
    >
      <small
        style={
          Object {
            "color": "#d1450d",
          }
        }
      >
        Trading on:
      </small>
      <h6>
        New York Stock Exchange Arca
      </h6>
    </div>
  </div>,
]
`;
